src/DocumentDB.ChangeFeedProcessor.UnitTests/PartitionManagement/PartitionControllerSplitTests.cs

```diff
@@ -15,211 +15,202 @@ namespace Microsoft.Azure.Documents.ChangeFeedProcessor.UnitTests.PartitionManag
     using Xunit;
 
     [Trait("Category", "Gated")]
-    public class PartitionControllerSplitTests : IAsyncLifetime
+    public class PartitionControllerSplitTests
     {
         private const string LastContinuationToken = "lastContinuation";
         private const string InitialContinuationToken = "initial token";
         private const string PartitionId = "partitionId";
-        private readonly ILease lease, leaseChild, leaseChild2;
-        private readonly ILeaseManager leaseManager;
-        private readonly IPartitionSupervisorFactory partitionSupervisorFactory;
-        private readonly IPartitionSynchronizer synchronizer;
-        private readonly PartitionController sut;
 
-        public PartitionControllerSplitTests()
+        private ILease MockLease(string partitionId = null)
         {
-            lease = Mock.Of<ILease>(l => l.PartitionId == PartitionId && l.ContinuationToken == InitialContinuationToken);
+            partitionId = partitionId ?? Guid.NewGuid().ToString();
+            return Mock.Of<ILease>(l => l.PartitionId == partitionId);
+        }
 
-            leaseChild = Mock.Of<ILease>();
-            Mock.Get(leaseChild)
-                .Setup(l => l.PartitionId)
-                .Returns("childPartitionId");
+        [Fact]
+        public async Task Controller_ShouldSignalSynchronizerSplitPartition_IfPartitionSplitHappened()
+        {
+            //arrange
+            var lease = MockLease(PartitionId);
+            var synchronizer = Mock.Of<IPartitionSynchronizer>();
+            Mock.Get(synchronizer)
+                .Setup(s => s.SplitPartitionAsync(lease))
+                .ReturnsAsync(new[] { MockLease(), MockLease() });
 
-            leaseChild2 = Mock.Of<ILease>();
-            Mock.Get(leaseChild2)
-                .Setup(l => l.PartitionId)
-                .Returns("childPartitionId2");
+            var partitionSupervisor = Mock.Of<IPartitionSupervisor>(o => o.RunAsync(It.IsAny<CancellationToken>()) == Task.FromException(new PartitionSplitException("message", LastContinuationToken)));
```

Comment:
Looks like these 3 lines are repeated, with some variation. Consider a helper method.

Reply:
I was trying but then the test will be less readable as you need to jump up/down to the method + I'd like to show the reader of the code that RynAsync is throwing exception .. so keeping it as it is .. we can change it in future .